include/nbl/video/SPhysicalDeviceFeatures.h

```diff
@@ -0,0 +1,102 @@
+#ifndef _NBL_VIDEO_S_PHYSICAL_DEVICE_FEATURES_H_INCLUDED_
+#define _NBL_VIDEO_S_PHYSICAL_DEVICE_FEATURES_H_INCLUDED_
+
+namespace nbl::video
+{
+
+struct SPhysicalDeviceFeatures
+{
+    /* Vulkan Core 1.0 */
+    bool robustBufferAccess = false;
+    //VkBool32    fullDrawIndexUint32;
+    bool imageCubeArray = false;
+    //VkBool32    independentBlend;
+    bool geometryShader    = false;
+    //VkBool32    tessellationShader;
+    //VkBool32    sampleRateShading;
+    //VkBool32    dualSrcBlend;
+    bool logicOp = false;
+    bool multiDrawIndirect = false;
+    //VkBool32    drawIndirectFirstInstance;
+    //VkBool32    depthClamp;
+    //VkBool32    depthBiasClamp;
+    //VkBool32    fillModeNonSolid;
+    //VkBool32    depthBounds;
+    //VkBool32    wideLines;
+    //VkBool32    largePoints;
+    //VkBool32    alphaToOne;
+    bool multiViewport = false;
+    bool samplerAnisotropy = false;
+    //VkBool32    textureCompressionETC2;
+    //VkBool32    textureCompressionASTC_LDR;
+    //VkBool32    textureCompressionBC;
```

Comment:
want to expose, because want to flag as enabled.

Reply:
yes

Reply:
by  `flag as enabled` do you mean `report as enabled` or `enable by default` for logical device?

Reply:
I mean I want the boolean to be able to enable the format support on Vulkan if the impl supports it

Reply:
What about impl on GL:
I think enable `textureCompressionASTC_LDR` when [KHR_texture_compression_astc_hdr](https://www.khronos.org/registry/OpenGL/extensions/KHR/KHR_texture_compression_astc_hdr.txt) is available (?) seems to have `GL_KHR_texture_compression_astc_ldr` in it's namestrings too

I couldn't find anything else for the other formats. should I get format support of their respective formats? like `EF_BC1_RGB_UNORM_BLOCK` for `BC`

I just remembered we never wrote the Format support stuff for OpenGL

Reply:
> I think enable textureCompressionASTC_LDR when [KHR_texture_compression_astc_hdr](https://www.khronos.org/registry/OpenGL/extensions/KHR/KHR_texture_compression_astc_hdr.txt) is available (?) seems to have GL_KHR_texture_compression_astc_ldr in it's namestrings too

whenever a GL extension page declares two name strings (its to keep the spec modification in one place), it means there's two separate namestrings to query, not that the extension will provide both whenever the main one is present.

Reply:
> I couldn't find anything else for the other formats. should I get format support of their respective formats? like EF_BC1_RGB_UNORM_BLOCK for BC

I think I wrote down in another comment how to detect those.

If you can't find it, we'll revisit at the end, since there's BC1-5 which is the S3TC compression, and then BC6 and 7 which is a different thing

Reply:
eh, we'll nuke these variables anyway.

Reply:
Yep :D